config/prow/cluster/label_sync_cron_job.yaml

```diff
@@ -16,11 +16,11 @@ spec:
         spec:
           containers:
             - name: label-sync
-              image: gcr.io/k8s-prow/label_sync:v20200724-4a9ed14895
+              image: gcr.io/k8s-prow/label_sync:v20210830-fcbb70627f
               args:
                 - --config=/etc/config/labels.yaml
                 - --confirm=true
-                - --only=ppc64le-cloud/test-infra,ocp-power-automation/ocp4-upi-powervs,ocp-power-automation/infra,ppc64le-cloud/kubetest2-plugins
+                - --only=ppc64le-cloud/test-infra,ocp-power-automation/ocp4-upi-powervs,ocp-power-automation/infra,ppc64le-cloud/kubetest2-plugins,ppc64le-cloud/k8s-ansible,ppc64le-cloud/pvsadm
```

Comment:
lets use `--org` option instead.

Reply:
I tried by including ```--orgs=ppc64le-cloud,ocp-power-automation,openshift-powervs```. (orgs mentioned for trigger in plugins)
Errors of below kind are noticed in label_sync pod log:
```
{"component":"label_sync","error":"failed to list labels: [status code 404 not one of [200], body: {\"message\":\"Not Found\",\"documentation_url\":\"https://docs.github.com/rest/reference/issues#update-a-label\"}
```
The error message does not mention which repo it faced 404 Not Found.
@mkumatag Can this failures be due to permissions issues.

Reply:
> I tried by including `--orgs=ppc64le-cloud,ocp-power-automation,openshift-powervs`. (orgs mentioned for trigger in plugins)
> Errors of below kind are noticed in label_sync pod log:
> 
> ```
> {"component":"label_sync","error":"failed to list labels: [status code 404 not one of [200], body: {\"message\":\"Not Found\",\"documentation_url\":\"https://docs.github.com/rest/reference/issues#update-a-label\"}
> ```
> 
> The error message does not mention which repo it faced 404 Not Found.
> @mkumatag Can this failures be due to permissions issues.

it is hard debug without the actual request information.

Reply:
Yes, the rate-limiting related errors appear in the log. I see 
```
{"client":"github","component":"label_sync","file":"prow/github/client.go:417","func":"k8s.io/test-infra/prow/github.(*throttler).Wait.func1","level":"warning","msg":"Throttled clientside for more than 15 minutes","severity":"warning","throttle-duration":"22m58.624348768s","throttled":true,"time":"2021-09-29T07:00:19Z"}
```
Any clue on how to fix this ratelimiting?
Also, the pod is erroring out with 

```
"component":"label_sync","error":"failed to list labels:  [status code 404 not one of [201], body: {\"message\":\"Not Found\",\"documentation_url\":\"https://docs.github.com/rest/reference/issues#create-a-label\"} .......
"file":"label_sync/main.go:788","func":"main.main","level":"fatal","msg":"failed to update ppc64le-cloud","severity":"fatal","time":"2021-09-29T07:16:19Z"}
```
@mkumatag can you please check if the bot has required access to `ppc64le-cloud/test-prow-bot` repo.

Reply:
> Yes, the rate-limiting related errors appear in the log. I see
> 
> ```
> {"client":"github","component":"label_sync","file":"prow/github/client.go:417","func":"k8s.io/test-infra/prow/github.(*throttler).Wait.func1","level":"warning","msg":"Throttled clientside for more than 15 minutes","severity":"warning","throttle-duration":"22m58.624348768s","throttled":true,"time":"2021-09-29T07:00:19Z"}
> ```
> 
> Any clue on how to fix this ratelimiting? Also, the pod is erroring out with
> 
> ```
> "component":"label_sync","error":"failed to list labels:  [status code 404 not one of [201], body: {\"message\":\"Not Found\",\"documentation_url\":\"https://docs.github.com/rest/reference/issues#create-a-label\"} .......
> "file":"label_sync/main.go:788","func":"main.main","level":"fatal","msg":"failed to update ppc64le-cloud","severity":"fatal","time":"2021-09-29T07:16:19Z"}
> ```
> 
> @mkumatag can you please check if the bot has required access to `ppc64le-cloud/test-prow-bot` repo.

not sure, didn't spend enough time to understand how community folks are dealing with this, need to talk to community folks

Reply:
I added the `ghproxy` related endpoints and a few more repos.
I don't see any rate-limiting errors. Tested it by syncing labels for the newly added `ocp-power-automation/ocp4-playbooks` repo.